sympy/integrals/manualintegrate.py

```diff
@@ -486,6 +487,56 @@ def _add_degenerate_step(generic_cond, generic_step, degenerate_step):
     return PiecewiseRule(subfunctions, generic_step.context, generic_step.symbol)
 
 
+def nested_pow_rule(integral: IntegralInfo):
+    # nested (c*(a+b*x)**d)**e
+    integrand, x = integral
+
+    a_ = Wild('a', exclude=[x])
+    b_ = Wild('b', exclude=[x, 0])
+    pattern = a_+b_*x
+    generic_cond = S.true
+
+    def _get_base_exp(expr: Expr) -> tuple[Expr, Expr]:
+        if not expr.has_free(x):
+            return S.One, S.Zero
+        if expr.is_Mul:
+            _, terms = expr.as_coeff_mul()
+            if not terms:
+                return S.One, S.Zero
+            results = [_get_base_exp(term) for term in terms]
+            bases = set(b for b, _ in results)
+            bases.discard(S.One)
+            if len(bases) == 1:
+                return bases.pop(), Add(*(e for _, e in results))
+            raise ValueError()
+        if expr.is_Pow:
+            b, e = expr.base, expr.exp  # type: ignore
+            if e.has_free(x):
+                raise ValueError()
+            base_, sub_exp = _get_base_exp(b)
+            return base_, sub_exp * e
```

Comment:
Does this assume that `(a**b**c) == a**(b*c)`? If so then that identity does not always hold.

Reply:
Basic power rule: to calculate integral of `x**n`, `n != -1`, we multiply it by `x` and divide it by `n+1`, so we get `x**(n+1)/(n+1)`.
Basic log rule: to calculate integral of `x**-1`, we multiply it by `x*log(x)`, so we get `log(x)`.

Now let's extend these basic rules to more complicated forms like  `(x**2)**(-1/2)` and `x*cbrt(1/x)`.

I know `(x**2)**(-1/2) != x**-1`, but to calculate the integral, we should decide whether to use power rule or log rule. The decision is made by treating `(x**2)**(-1/2)` as `x**(2*(-1/2))`, so the "nominal" exponent is -1, thus we apply log rule: multiply it by `x*log(x)` to get `x*log(x)/sqrt(x**2)`. The correctness is easily verified: `diff(x*log(x)/sqrt(x**2)) == 1/sqrt(x**2)`.

Similar thing happens to `x*cbrt(1/x)` as well. The "nominal" exponent is `1+(1/3)*(-1) == 2/3` which is not -1, so we apply power rule: multiply it by `x` and divide it by `5/3`. The result is `3*x**2*cbrt(1/x)/5`, and `diff(3*x**2*cbrt(1/x)/5) == x*(1/x)**(1/3)`.

Reply:
Oh, I see. So we just use `sub_exp*e` in the divisor.